// together-ai/lib/node-unsafe-imports.js
"use strict";
/*
 * Importing node APIs breaks the SDK in the browser.
 * The ideal scenario is that we do not use nodejs exclusive APIs in this codebase
 *
 * In the interim, this file provides a way to import the APIs in a way that will not break the SDK in the browser.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.readFile = exports.resolve = exports.isUtf8 = exports.readline = exports.extname = exports.createReadStream = exports.stat = void 0;
let statMethod;
let resolveMethod;
let createReadStreamMethod;
let extnameMethod;
let readlineMethod;
let isUtf8Method;
let readFileMethod;
try {
    exports.stat = statMethod = require('fs/promises').stat;
    exports.createReadStream = createReadStreamMethod = require('fs').createReadStream;
    exports.extname = extnameMethod = require('path').extname;
    exports.readline = readlineMethod = require('readline');
    exports.isUtf8 = isUtf8Method = require('buffer').isUtf8;
    exports.resolve = resolveMethod = require('path').resolve;
    exports.readFile = readFileMethod = require('fs/promises').readFile;
}
catch { }
