// together-ai/resources/beta/jig/jig.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Jig = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const QueueAPI = tslib_1.__importStar(require("./queue.js"));
const queue_1 = require("./queue.js");
const SecretsAPI = tslib_1.__importStar(require("./secrets.js"));
const secrets_1 = require("./secrets.js");
const VolumesAPI = tslib_1.__importStar(require("./volumes.js"));
const volumes_1 = require("./volumes.js");
const path_1 = require("../../../internal/utils/path.js");
class Jig extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.queue = new QueueAPI.Queue(this._client);
        this.volumes = new VolumesAPI.Volumes(this._client);
        this.secrets = new SecretsAPI.Secrets(this._client);
    }
    /**
     * Retrieve details of a specific deployment by its ID or name
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/deployments/${id}`, options);
    }
    /**
     * Update an existing deployment configuration
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/deployments/${id}`, { body, ...options });
    }
    /**
     * Get a list of all deployments in your project
     */
    list(options) {
        return this._client.get('/deployments', options);
    }
    /**
     * Create a new deployment with specified configuration
     */
    deploy(body, options) {
        return this._client.post('/deployments', { body, ...options });
    }
    /**
     * Delete an existing deployment
     */
    destroy(id, options) {
        return this._client.delete((0, path_1.path) `/deployments/${id}`, options);
    }
    /**
     * Retrieve logs from a deployment, optionally filtered by replica ID.
     */
    retrieveLogs(id, query = {}, options) {
        return this._client.get((0, path_1.path) `/deployments/${id}/logs`, { query, ...options });
    }
}
exports.Jig = Jig;
Jig.Queue = queue_1.Queue;
Jig.Volumes = volumes_1.Volumes;
Jig.Secrets = secrets_1.Secrets;
