// together-ai/resources/beta/jig/jig.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as QueueAPI from "./queue.mjs";
import { Queue, } from "./queue.mjs";
import * as SecretsAPI from "./secrets.mjs";
import { Secrets, } from "./secrets.mjs";
import * as VolumesAPI from "./volumes.mjs";
import { Volumes, } from "./volumes.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class Jig extends APIResource {
    constructor() {
        super(...arguments);
        this.queue = new QueueAPI.Queue(this._client);
        this.volumes = new VolumesAPI.Volumes(this._client);
        this.secrets = new SecretsAPI.Secrets(this._client);
    }
    /**
     * Retrieve details of a specific deployment by its ID or name
     */
    retrieve(id, options) {
        return this._client.get(path `/deployments/${id}`, options);
    }
    /**
     * Update an existing deployment configuration
     */
    update(id, body, options) {
        return this._client.patch(path `/deployments/${id}`, { body, ...options });
    }
    /**
     * Get a list of all deployments in your project
     */
    list(options) {
        return this._client.get('/deployments', options);
    }
    /**
     * Create a new deployment with specified configuration
     */
    deploy(body, options) {
        return this._client.post('/deployments', { body, ...options });
    }
    /**
     * Delete an existing deployment
     */
    destroy(id, options) {
        return this._client.delete(path `/deployments/${id}`, options);
    }
    /**
     * Retrieve logs from a deployment, optionally filtered by replica ID.
     */
    retrieveLogs(id, query = {}, options) {
        return this._client.get(path `/deployments/${id}/logs`, { query, ...options });
    }
}
Jig.Queue = Queue;
Jig.Volumes = Volumes;
Jig.Secrets = Secrets;
