// together-ai/resources/beta/jig/secrets.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class Secrets extends APIResource {
    /**
     * Create a new secret to store sensitive configuration values
     */
    create(body, options) {
        return this._client.post('/deployments/secrets', { body, ...options });
    }
    /**
     * Retrieve details of a specific secret by its ID or name
     */
    retrieve(id, options) {
        return this._client.get(path `/deployments/secrets/${id}`, options);
    }
    /**
     * Update an existing secret's value or metadata
     */
    update(id, body, options) {
        return this._client.patch(path `/deployments/secrets/${id}`, { body, ...options });
    }
    /**
     * Retrieve all secrets in your project
     */
    list(options) {
        return this._client.get('/deployments/secrets', options);
    }
    /**
     * Delete an existing secret
     */
    delete(id, options) {
        return this._client.delete(path `/deployments/secrets/${id}`, options);
    }
}
