// together-ai/resources/beta/jig/secrets.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Secrets = void 0;
const resource_1 = require("../../../core/resource.js");
const path_1 = require("../../../internal/utils/path.js");
class Secrets extends resource_1.APIResource {
    /**
     * Create a new secret to store sensitive configuration values
     */
    create(body, options) {
        return this._client.post('/deployments/secrets', { body, ...options });
    }
    /**
     * Retrieve details of a specific secret by its ID or name
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/deployments/secrets/${id}`, options);
    }
    /**
     * Update an existing secret's value or metadata
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/deployments/secrets/${id}`, { body, ...options });
    }
    /**
     * Retrieve all secrets in your project
     */
    list(options) {
        return this._client.get('/deployments/secrets', options);
    }
    /**
     * Delete an existing secret
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/deployments/secrets/${id}`, options);
    }
}
exports.Secrets = Secrets;
