// together-ai/resources/beta/jig/volumes.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Volumes = void 0;
const resource_1 = require("../../../core/resource.js");
const path_1 = require("../../../internal/utils/path.js");
class Volumes extends resource_1.APIResource {
    /**
     * Create a new volume to preload files in deployments
     */
    create(body, options) {
        return this._client.post('/deployments/storage/volumes', { body, ...options });
    }
    /**
     * Retrieve details of a specific volume by its ID or name
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/deployments/storage/volumes/${id}`, options);
    }
    /**
     * Update an existing volume's configuration or contents
     */
    update(id, body, options) {
        return this._client.patch((0, path_1.path) `/deployments/storage/volumes/${id}`, { body, ...options });
    }
    /**
     * Retrieve all volumes in your project
     */
    list(options) {
        return this._client.get('/deployments/storage/volumes', options);
    }
    /**
     * Delete an existing volume
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/deployments/storage/volumes/${id}`, options);
    }
}
exports.Volumes = Volumes;
