// together-ai/resources/beta/jig/volumes.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class Volumes extends APIResource {
    /**
     * Create a new volume to preload files in deployments
     */
    create(body, options) {
        return this._client.post('/deployments/storage/volumes', { body, ...options });
    }
    /**
     * Retrieve details of a specific volume by its ID or name
     */
    retrieve(id, options) {
        return this._client.get(path `/deployments/storage/volumes/${id}`, options);
    }
    /**
     * Update an existing volume's configuration or contents
     */
    update(id, body, options) {
        return this._client.patch(path `/deployments/storage/volumes/${id}`, { body, ...options });
    }
    /**
     * Retrieve all volumes in your project
     */
    list(options) {
        return this._client.get('/deployments/storage/volumes', options);
    }
    /**
     * Delete an existing volume
     */
    delete(id, options) {
        return this._client.delete(path `/deployments/storage/volumes/${id}`, options);
    }
}
