// together-ai/resources/beta/clusters/clusters.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../core/resource.mjs";
import * as StorageAPI from "./storage.mjs";
import { Storage, } from "./storage.mjs";
import { path } from "../../../internal/utils/path.mjs";
export class Clusters extends APIResource {
    constructor() {
        super(...arguments);
        this.storage = new StorageAPI.Storage(this._client);
    }
    /**
     * Create an Instant Cluster on Together's high-performance GPU clusters. With
     * features like on-demand scaling, long-lived resizable high-bandwidth shared
     * DC-local storage, Kubernetes and Slurm cluster flavors, a REST API, and
     * Terraform support, you can run workloads flexibly without complex infrastructure
     * management.
     */
    create(body, options) {
        return this._client.post('/compute/clusters', { body, ...options });
    }
    /**
     * Retrieve information about a specific GPU cluster.
     */
    retrieve(clusterID, options) {
        return this._client.get(path `/compute/clusters/${clusterID}`, options);
    }
    /**
     * Update the configuration of an existing GPU cluster.
     */
    update(clusterID, body, options) {
        return this._client.put(path `/compute/clusters/${clusterID}`, { body, ...options });
    }
    /**
     * List all GPU clusters.
     */
    list(options) {
        return this._client.get('/compute/clusters', options);
    }
    /**
     * Delete a GPU cluster by cluster ID.
     */
    delete(clusterID, options) {
        return this._client.delete(path `/compute/clusters/${clusterID}`, options);
    }
    /**
     * List regions and corresponding supported driver versions
     */
    listRegions(options) {
        return this._client.get('/compute/regions', options);
    }
}
Clusters.Storage = Storage;
