// together-ai/resources/beta/clusters/clusters.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Clusters = void 0;
const tslib_1 = require("../../../internal/tslib.js");
const resource_1 = require("../../../core/resource.js");
const StorageAPI = tslib_1.__importStar(require("./storage.js"));
const storage_1 = require("./storage.js");
const path_1 = require("../../../internal/utils/path.js");
class Clusters extends resource_1.APIResource {
    constructor() {
        super(...arguments);
        this.storage = new StorageAPI.Storage(this._client);
    }
    /**
     * Create an Instant Cluster on Together's high-performance GPU clusters. With
     * features like on-demand scaling, long-lived resizable high-bandwidth shared
     * DC-local storage, Kubernetes and Slurm cluster flavors, a REST API, and
     * Terraform support, you can run workloads flexibly without complex infrastructure
     * management.
     */
    create(body, options) {
        return this._client.post('/compute/clusters', { body, ...options });
    }
    /**
     * Retrieve information about a specific GPU cluster.
     */
    retrieve(clusterID, options) {
        return this._client.get((0, path_1.path) `/compute/clusters/${clusterID}`, options);
    }
    /**
     * Update the configuration of an existing GPU cluster.
     */
    update(clusterID, body, options) {
        return this._client.put((0, path_1.path) `/compute/clusters/${clusterID}`, { body, ...options });
    }
    /**
     * List all GPU clusters.
     */
    list(options) {
        return this._client.get('/compute/clusters', options);
    }
    /**
     * Delete a GPU cluster by cluster ID.
     */
    delete(clusterID, options) {
        return this._client.delete((0, path_1.path) `/compute/clusters/${clusterID}`, options);
    }
    /**
     * List regions and corresponding supported driver versions
     */
    listRegions(options) {
        return this._client.get('/compute/regions', options);
    }
}
exports.Clusters = Clusters;
Clusters.Storage = storage_1.Storage;
