// together-ai/resources/beta/clusters/storage.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Storage = void 0;
const resource_1 = require("../../../core/resource.js");
const path_1 = require("../../../internal/utils/path.js");
class Storage extends resource_1.APIResource {
    /**
     * Instant Clusters supports long-lived, resizable in-DC shared storage with user
     * data persistence. You can dynamically create and attach volumes to your cluster
     * at cluster creation time, and resize as your data grows. All shared storage is
     * backed by multi-NIC bare metal paths, ensuring high-throughput and low-latency
     * performance for shared storage.
     */
    create(body, options) {
        return this._client.post('/compute/clusters/storage/volumes', { body, ...options });
    }
    /**
     * Retrieve information about a specific shared volume.
     */
    retrieve(volumeID, options) {
        return this._client.get((0, path_1.path) `/compute/clusters/storage/volumes/${volumeID}`, options);
    }
    /**
     * Update the configuration of an existing shared volume.
     */
    update(body, options) {
        return this._client.put('/compute/clusters/storage/volumes', { body, ...options });
    }
    /**
     * List all shared volumes.
     */
    list(options) {
        return this._client.get('/compute/clusters/storage/volumes', options);
    }
    /**
     * Delete a shared volume. Note that if this volume is attached to a cluster,
     * deleting will fail.
     */
    delete(volumeID, options) {
        return this._client.delete((0, path_1.path) `/compute/clusters/storage/volumes/${volumeID}`, options);
    }
}
exports.Storage = Storage;
