// together-ai/resources/files.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Files = void 0;
const resource_1 = require("../core/resource.js");
const headers_1 = require("../internal/headers.js");
const path_1 = require("../internal/utils/path.js");
const upload_1 = require("../lib/upload.js");
class Files extends resource_1.APIResource {
    /**
     * Retrieve the metadata for a single uploaded data file.
     *
     * @example
     * ```ts
     * const fileResponse = await client.files.retrieve('id');
     * ```
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/files/${id}`, options);
    }
    /**
     * List the metadata for all uploaded data files.
     *
     * @example
     * ```ts
     * const fileList = await client.files.list();
     * ```
     */
    list(options) {
        return this._client.get('/files', options);
    }
    /**
     * Delete a previously uploaded data file.
     *
     * @example
     * ```ts
     * const file = await client.files.delete('id');
     * ```
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/files/${id}`, options);
    }
    /**
     * Get the contents of a single uploaded data file.
     *
     * @example
     * ```ts
     * const response = await client.files.content('id');
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    content(id, options) {
        return this._client.get((0, path_1.path) `/files/${id}/content`, {
            ...options,
            headers: (0, headers_1.buildHeaders)([{ Accept: 'application/binary' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Upload a file.
     * CUSTOM CODE. DO NOT LET STAINLESS REMOVE.
     */
    upload(file, purpose, check = true) {
        return (0, upload_1.upload)(this._client, file, purpose, check);
    }
}
exports.Files = Files;
