// together-ai/resources/fine-tuning.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../core/resource.mjs";
import { buildHeaders } from "../internal/headers.mjs";
import { path } from "../internal/utils/path.mjs";
export class FineTuning extends APIResource {
    /**
     * Create a fine-tuning job with the provided model and training data.
     *
     * @example
     * ```ts
     * const fineTuning = await client.fineTuning.create({
     *   model: 'model',
     *   training_file: 'training_file',
     * });
     * ```
     */
    create(body, options) {
        return this._client.post('/fine-tunes', { body, ...options });
    }
    /**
     * List the metadata for a single fine-tuning job.
     *
     * @example
     * ```ts
     * const finetuneResponse = await client.fineTuning.retrieve(
     *   'id',
     * );
     * ```
     */
    retrieve(id, options) {
        return this._client.get(path `/fine-tunes/${id}`, options);
    }
    /**
     * List the metadata for all fine-tuning jobs. Returns a list of
     * FinetuneResponseTruncated objects.
     *
     * @example
     * ```ts
     * const fineTunings = await client.fineTuning.list();
     * ```
     */
    list(options) {
        return this._client.get('/fine-tunes', options);
    }
    /**
     * Delete a fine-tuning job.
     *
     * @example
     * ```ts
     * const fineTuning = await client.fineTuning.delete('id');
     * ```
     */
    delete(id, params = {}, options) {
        const { force } = params ?? {};
        return this._client.delete(path `/fine-tunes/${id}`, { query: { force }, ...options });
    }
    /**
     * Cancel a currently running fine-tuning job. Returns a FinetuneResponseTruncated
     * object.
     *
     * @example
     * ```ts
     * const response = await client.fineTuning.cancel('id');
     * ```
     */
    cancel(id, options) {
        return this._client.post(path `/fine-tunes/${id}/cancel`, options);
    }
    /**
     * Receive a compressed fine-tuned model or checkpoint.
     *
     * @example
     * ```ts
     * const response = await client.fineTuning.content({
     *   ft_id: 'ft_id',
     * });
     *
     * const content = await response.blob();
     * console.log(content);
     * ```
     */
    content(query, options) {
        return this._client.get('/finetune/download', {
            query,
            ...options,
            headers: buildHeaders([{ Accept: 'application/octet-stream' }, options?.headers]),
            __binaryResponse: true,
        });
    }
    /**
     * Estimate the price of a fine-tuning job.
     *
     * @example
     * ```ts
     * const response = await client.fineTuning.estimatePrice({
     *   training_file: 'training_file',
     * });
     * ```
     */
    estimatePrice(body, options) {
        return this._client.post('/fine-tunes/estimate-price', { body, ...options });
    }
    /**
     * List the checkpoints for a single fine-tuning job.
     *
     * @example
     * ```ts
     * const response = await client.fineTuning.listCheckpoints(
     *   'id',
     * );
     * ```
     */
    listCheckpoints(id, options) {
        return this._client.get(path `/fine-tunes/${id}/checkpoints`, options);
    }
    /**
     * List the events for a single fine-tuning job.
     *
     * @example
     * ```ts
     * const response = await client.fineTuning.listEvents('id');
     * ```
     */
    listEvents(id, options) {
        return this._client.get(path `/fine-tunes/${id}/events`, options);
    }
}
